vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceTextBlockController.java

```diff
@@ -0,0 +1,72 @@
+package edu.asu.diging.vspace.web.staff;
+
+import java.io.IOException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import edu.asu.diging.vspace.core.exception.ImageCouldNotBeStoredException;
+import edu.asu.diging.vspace.core.exception.SpaceDoesNotExistException;
+import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.model.display.ISpaceTextBlockDisplay;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+import edu.asu.diging.vspace.core.services.ISpaceTextBlockManager;
+
+@Controller
+public class AddSpaceTextBlockController {
+
+    @Autowired
+    private ISpaceManager spaceManager;
+
+    @Autowired
+    private ISpaceTextBlockManager spaceTextBlockManager;
+
+
+    @RequestMapping(value = "/staff/space/{id}/textBlock", method = RequestMethod.POST)
+    public ResponseEntity<String> createSpaceTextBlock(@PathVariable("id") String id, @RequestParam("x") String x,
+            @RequestParam("y") String y, @RequestParam("textContent") String text,
+            @RequestParam("height") String height, @RequestParam("width") String width)
+                    throws NumberFormatException, SpaceDoesNotExistException, IOException, ImageCouldNotBeStoredException {
```

Comment:
where is `NumberFormatException` thrown and why? you should probably not throw that but handle it if it's actually thrown.

Reply:
pretty much the same with all the exceptions